src/Native/jitinterface/jitwrapper.cpp

```diff
@@ -0,0 +1,44 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+#include "corinfoexception.h"
+
+class Jit
+{
+public:
+    virtual int compileMethod(
+        void* compHnd,
+        void* methodInfo,
+        unsigned flags,
+        void* entryAddress,
+        void* nativeSizeOfCode) = 0;
+};
+
+extern "C" int JitWrapper(
+    CorInfoException **ppException,
+    Jit* pJit,
+    void* compHnd,
+    void* methodInfo,
+    unsigned flags,
+    void* entryAddress,
+    void* nativeSizeOfCode)
+{
+    *ppException = nullptr;
+
+    try
+    {
+        return pJit->compileMethod(compHnd, methodInfo, flags, entryAddress, nativeSizeOfCode);
+    }
+    catch (CorInfoException *pException)
+    {
+        *ppException = pException;
+    }
+    catch (...)
```

Comment:
It this needed for anything? We should not be catching somebody else's exceptions.